src/main/java/com/customer/controller/CustomerController.java

```diff
@@ -24,31 +31,49 @@
 	private CustomerServiceImpl customerServiceImpl;
 
 	@Autowired
-	CustomerController(CustomerServiceImpl customerServiceImpl)
-	{
+	CustomerController(CustomerServiceImpl customerServiceImpl) {
 		this.customerServiceImpl = customerServiceImpl;
 	}
 
-	// get the list of all the customers
+	/* get the list of all the customers */
 	@GetMapping("/customers")
 	public List<CustomerResponse> getAllCustomers() {
-
 		return customerServiceImpl.getAllCustomers();
-
 	}
 
 	/* Post request */
-	@PostMapping("/customers")
-	public String createCustomer(@Valid @RequestBody Customer customer) {
-		customerServiceImpl.createCustomer(customer);
-		return "Customer is created successfully";
-
+	@PostMapping("/customerRegister")
+	public ResponseEntity<Void> createCustomer(@Valid @RequestBody Customer customer) throws InvalidRequest {
+		if (customer != null) {
+			customerServiceImpl.createCustomer(customer);
+			return ResponseEntity.ok().build();
+		} else
+			throw new ForbiddenException("Please provide valid credentials");
 	}
 
 	/* get the customer by Id */
 	@GetMapping("/customers/{id}")
-	public @ResponseBody CustomerResponse findCustomerById(@PathVariable int id) {
-		return customerServiceImpl.findCustomerById(id);
+	public @ResponseBody CustomerResponse findCustomerById(@PathVariable int id, Principal principal) {
+		return customerServiceImpl.findCustomerById(id, principal);
+	}
+
+	/* update the customer details */
+	@PutMapping("/customers/{id}")
+	public ResponseEntity<Void> updateCustomer(@PathVariable int id, @RequestBody Customer customer,
+			Principal principal) {
+		customerServiceImpl.updateCustomerDetails(id, customer, principal);
+		return ResponseEntity.ok().build();
 	}
 
+	/* Delete the customer details */
+	@DeleteMapping("/customers/{id}")
+	public void DeleteCustomerDetails(@PathVariable int id, Principal principal) {
+		customerServiceImpl.deleteCustomer(id, principal);
+	}
+
+	/* get the customer by Id */
+	@GetMapping("/customer/{id}")
+	public @ResponseBody Optional<Customer> findCustomerById(@PathVariable int id) {
```

Comment:
Return a `REsponseEntity` with appropriate status here too, instead of Customer directly

Reply:
Done.